### PR TITLE
fix ancient block

### PR DESCRIPTION
**Pull request by blockchaindevsh**

### CODE REVIEW

File: core/rawdb/accessors_chain.go [hunk 1]
```diff
@@ -337,8 +336,12 @@ func ReadHeaderRLP(db ethdb.Reader, hash common.Hash, number uint64) rlp.RawValu
 		// comparison is necessary since ancient database only maintains
 		// the canonical data.
 		data, _ = reader.Ancient(freezerHeaderTable, number)
-		if len(data) > 0 && crypto.Keccak256Hash(data) == hash {
-			return nil
+		if len(data) > 0 {
+			var header *types.Header
+			rlp.DecodeBytes(data, &header)
```
Comment (qizhou) [inline core/rawdb/accessors_chain.go:341]: Should we check the error of this function call?

Comment (blockchaindevsh) [reply core/rawdb/accessors_chain.go]: Yeah, missed out the error handling, added.

File: core/rawdb/accessors_chain.go [hunk 2]
```diff
@@ -332,18 +331,28 @@ func ReadHeaderRange(db ethdb.Reader, number uint64, count uint64) []rlp.RawValu
...
+			if header != nil && header.Hash() == hash {
+				return nil
+			}
 		}
 		// If not, try reading from leveldb
 		data, _ = db.Get(headerKey(number, hash))
 		return nil
 	})
+	if err != nil {
+		data = nil
```
Comment (qizhou) [inline core/rawdb/accessors_chain.go:354]: return nil directly?

**Review by qizhou — APPROVED** [review]

File: core/rawdb/accessors_chain.go (hunk 2 above)
Comment (blockchaindevsh) [reply core/rawdb/accessors_chain.go]: Done.

**Pull request merged**